.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,68 @@
+name: Build and Push
+
+# Builds and pushes the branch to ECR from master, or the manually selected branch when invoked manually. We stopped using Docker Hub for storing the image of oplogtoredis, where we used a Webhook to do the same thing.
+
+on:
+  push:
+    branches:
+      - master
+  # Left this here in-case the workflow needs to be developed further rapidly:
+  # pull_request:
+  #   types: [ opened, synchronize, reopened ]
+  workflow_dispatch:
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+permissions:
+  id-token: write # for the creds itself
```

Comment:
Remind me again, the meaning behind the comment `for the creds itself` ?

Reply:
Honestly, I had no idea whatever this was required or not and what the comment meant. It looks like it was written by Ho Man, or someone before him and it was copied around (like over here). I remember failing the login without setting these permissions there, so I guess I assumed that it's needed over here as well, but it looks like it's not. Never looked into it deep enough.

Reply:
id-token is required for oidc role assumption